.github/workflows/reusable_dockerfile_pipeline.yml

```
@@ -53,8 +83,8 @@ jobs:
 
   docker-build:
     runs-on: "ubuntu-latest"
-    # wait until the security scanner will be done
-    needs: docker-security
+    # wait until the jobs are finished.
+    needs: ["prepare-env","docker-security"]
```

this should do the trick I suppose
```suggestion
    needs: ["prepare-env", "docker-security"]
```